lang/python314-devel/files/patch-target-mm-malloc.diff

```diff
@@ -0,0 +1,37 @@
+There is no support for aligned_malloc() in macOS < 10.15. Instead of blocking SIMD, just fallback to using _mm_malloc().
+--- Modules/_hacl/include/krml/internal/target.h
++++ Modules/_hacl/include/krml/internal/target.h
+@@ -19,6 +19,11 @@
+ #  define inline __inline__
+ #endif
+ 
++/* Include Apple-specific macros for use in defining KRML_ALIGNED_MALLOC. */
++#if defined(__APPLE__) && defined(__MACH__)
++#include <AvailabilityMacros.h>
++#endif
++
+ /******************************************************************************/
+ /* Macros that KaRaMeL will generate.                                         */
+ /******************************************************************************/
+@@ -123,7 +128,8 @@
+ #endif
+ 
+ /* MinGW-W64 does not support C11 aligned_alloc, but it supports
+- * MSVC's _aligned_malloc.
++ * MSVC's _aligned_malloc. Also, fallback to use mm_malloc.h
++ * implementation for macOS systems prior to 10.15 Catalina.
+  */
+ #ifndef KRML_ALIGNED_MALLOC
+ #  ifdef __MINGW32__
+@@ -133,6 +139,11 @@
+       defined(_MSC_VER) ||                                                     \
+       (defined(__MINGW32__) && defined(__MINGW64_VERSION_MAJOR)))
+ #    define KRML_ALIGNED_MALLOC(X, Y) _aligned_malloc(Y, X)
++#  elif defined(__APPLE__) && defined(__MACH__) &&                             \
++        defined(MAC_OS_X_VERSION_MIN_REQUIRED) &&                              \
++        (MAC_OS_X_VERSION_MIN_REQUIRED < 101500)
++#    include <mm_malloc.h>
++#    define KRML_ALIGNED_MALLOC(X, Y) _mm_malloc(Y, X)
```

Comment:
Don't you need to use `_mm_free` on memory allocated by `_mm_malloc`? It's unfortunate that `posix_memalign` has a different interface that doesn't easily fit in here. But plain `malloc` might well be sufficient, since its allocations are guaranteed to be correctly aligned for SIMD data types on macOS.